joomla/joomla-cms
# [com_cache] Success and error messages

#### Description

When deleting cached items, the user performs and action,so  like in many other views, should receive a success or error message.

So, this PR:
- adds success messages when deleting cache
- adds error messages when an error occured when deleting cache

Also:
- removes deprecated JError usage from com_cache
- other code/coments improvements (check code changes)
#### How to test
1. Use latest staging
2. Go to System -> Global configuration and enable some type of caching
3. Navigate in the backend and in the frontend to generate some cache
4. Go to System -> Clear Cache, you will see some cached items in "site" and in "administrator"
5. Delete some items, you will see you get no message after the page reloads.
6. Apply this patch
7. Repeat step 5. You will get a success message.

For testing error messages you need to simulate an error (like changing a cache group folder permissions in /cache/).
